testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/management/persistence/RemoteSshGitRepositoryTestCase.java

```diff
@@ -171,10 +174,19 @@ public static void setUp() throws Exception {
 
     }
 
+    static void backupConfiguration() throws IOException {
+        Path backUpRoot = Files.createTempDirectory("BackUpConfigurationFiles").resolve("configuration");
+        Files.createDirectories(backUpRoot);
+        PathUtil.copyRecursively(getJbossServerBaseDir().resolve("configuration"), backUpRoot, true);
+
+        RemoteSshGitRepositoryTestCase.backupRoot = backUpRoot;
+    }
+
     @AfterClass
     public static void afterClass() throws IOException {
         Security.removeProvider(CREDENTIAL_STORE_PROVIDER.getName());
         FileUtils.delete(CS_PUBKEY.toFile(), FileUtils.RECURSIVE | FileUtils.RETRY);
+        PathUtil.deleteRecursively(backupRoot);
```

Comment:
should we also restore the configuration in the `@AfterClass` to avoid keeping the config from the last test for the rest of the test suite?

Reply:
The configuration is automatically restored after each individual test on line 260 at the end of the tearDown method.

The @BeforeClass and @AfterClass methods are focused on taking a backup once at the start of the test run and deleting that backup at the end of the test run with the @After method handling the individual clean up after every test.